src/StoreKeeper/WooCommerce/B2C/Exports/OrderExport.php

```diff
@@ -394,6 +414,66 @@ public static function convertWooCommerceToStorekeeperOrderStatus($wc_order_stat
         }
     }
 
+    /**
+     * @param WC_Order $databaseOrder - Order items to compare from
+     * @param $backofficeOrder - Order items to compare to
+     */
+    public function checkOrderDifference(WC_Order $databaseOrder, $backofficeOrder): bool
+    {
+        $databaseOrderItems = $this->getOrderItems($databaseOrder);
+        $backofficeOrderItems = $backofficeOrder['order_items'];
+
+        $allHasExtras = true;
+        foreach ($backofficeOrderItems as $backofficeOrderItem) {
+            if (!array_key_exists('extra', $backofficeOrderItem)) {
+                $allHasExtras = false;
+                break;
+            }
+        }
+
+        if ($allHasExtras) {
+            $hasDifference = $this->checkOrderDifferenceByExtra($databaseOrderItems, $backofficeOrderItems);
+        } else {
+            $hasDifference = $this->checkOrderDifferenceBySet($databaseOrderItems, $backofficeOrderItems);
+        }
+
+        // In case order items are the same but prices have changed. e.g payment gateway fee
+        if (!$hasDifference && ((float) $databaseOrder->get_total()) !== $backofficeOrder['value_wt']) {
+            $hasDifference = true;
+        }
+
+        return $hasDifference;
+    }
+
+    private function checkOrderDifferenceByExtra(array $databaseOrderItems, array $backofficeOrderItems): bool
+    {
+        return array_column($databaseOrderItems, 'extra') !== array_column($backofficeOrderItems, 'extra');
```

Comment:
maybe sort the first? or are you 100% sure  they are always returned in the same order?

Reply:
Not sure about the order so I sorted.